lib/src/components/movement_sensor/movement_sensor.dart

```diff
@@ -52,54 +52,7 @@ abstract class MovementSensor extends Sensor {
   ///   compass: [double],
   ///   orientation: [Orientation],
   @override
-  Future<Map<String, dynamic>> readings({Map<String, dynamic>? extra}) async {
-    final Map<String, dynamic> readings = {};
-    try {
-      final Position pos = await position(extra: extra);
-      readings['position'] = pos.coordinates;
-      readings['altitude'] = pos.altitude;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-      print('Error: $exception');
-    }
-
-    try {
-      final lv = await linearVelocity(extra: extra);
-      readings['linear_velocity'] = lv;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-    }
-
-    try {
-      final av = await angularVelocity(extra: extra);
-      readings['angular_velocity'] = av;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-    }
-
-    try {
-      final la = await linearAcceleration(extra: extra);
-      readings['linear_acceleration'] = la;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-    }
-
-    try {
-      final comp = await compassHeading(extra: extra);
-      readings['compass'] = comp;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-    }
-
-    try {
-      final orient = await orientation(extra: extra);
-      readings['orientation'] = orient;
-    } catch (exception) {
-      // TODO: Check if the exception is of a specific type and ignore or rethrow
-    }
-
-    return readings;
-  }
```

Comment:
i _dont think_ this logic should have been here, moved to the client where i think it makes more sense with our current patterns.

Reply:
I think it should stay here, that way all subclasses of this resource will have this implementation out of the box (which I think is a good default)

Reply:
SGTM I will move this back.